src/ripple/server/impl/ServerHandlerImp.cpp

```diff
@@ -704,17 +707,39 @@ parse_Ports (BasicConfig const& config, std::ostream& log)
         result.push_back(to_Port(parsed, log));
     }
 
-    std::size_t count = 0;
-    for (auto const& p : result)
-        if (p.protocol.count("peer") > 0)
-            ++count;
-    if (count > 1)
+    if (!acceptPeers)
     {
-        log << "Error: More than one peer protocol configured in [server]\n";
-        Throw<std::exception> ();
+        auto it = result.begin ();
+
+        while (it != result.end())
+        {
+            auto& p = it->protocol;
+
+            // Remove the peer protocol, and if that would
+            // leave the port empty, remove the port as well
+            if (p.erase ("peer") && p.empty())
+                it = result.erase (it);
+            else
+                ++it;
+        }
+    }
+    else
+    {
+        std::size_t count = 0;
+        for (auto const& p : result)
+        {
+            if (p.protocol.count("peer"))
+                ++count;
+        }
```

Comment:
I think the `std::count_if` is neater - used it. I contemplated using a generic lambda but decided not to.